src/main/java/com/bootcamp/demo/FirebaseController.java

```diff
@@ -42,16 +53,67 @@ private void initFirestore() throws IOException {
         initializeApp(options);
 
         firestoreDB = getFirestore();
+        bookingService.setDb(firestoreDB);
     }
 
     /**
      * Returns the path for all collections stored in Firestore
      */
     @GetMapping("/getAllPaths")
     public Set<String> getAllPaths() {
-        return StreamSupport.stream(firestoreDB.listCollections().spliterator(), false)
+        return stream(firestoreDB.listCollections().spliterator(), false)
                 .map(CollectionReference::getPath)
                 .collect(Collectors.toUnmodifiableSet());
 
     }
+
+    @PostMapping("/createBooking")
+    public ResponseEntity<Object> createBooking(final Booking booking){
```

Comment:
Try to format the code, as it helps to keep the same structure. Here the missing space is minor, but it is good practice to get used to doing this.

Reply:
Also, should this be marked with `@RequestBody`?